adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,928 @@
+void launch_grouped_reduce(sycl::handler& h, sycl::range<3> r,
+                           sycl::range<3> size, const sycl::kernel& k,
+                           Args&&... args); (3)
+
+template <typename... Args>
+void launch_grouped_reduce(sycl::queue q, sycl::range<1> r,
+                           sycl::range<1> size, const sycl::kernel& k,
+                           Args&&... args); (4)
+
+template <typename... Args>
+void launch_grouped_reduce(sycl::queue q, sycl::range<2> r,
+                           sycl::range<2> size, const sycl::kernel& k,
+                           Args&&... args); (5)
+
+template <typename... Args>
+void launch_grouped_reduce(sycl::queue q, sycl::range<3> r,
+                           sycl::range<3> size, const sycl::kernel& k,
+                           Args&&... args); (6)
+
+}
+----
+_Effects_: Enqueues a kernel object that is invoked for every work-item in the
+specified [code]#sycl::range#, where each work-item contributes to one or more
+reductions.
+Work-items are grouped into work-groups of size [code]#size#.
+The arguments in [code]#args# are passed to the kernel in the same order.
+
+'''
+
+.[apititle]#launch_task# (kernel function)
+[source,role=synopsis,id=api:launch_task]
+----
+namespace sycl::khr {
+
+template <typename KernelType>
+void launch_task(sycl::handler& h, const KernelType& k); (1)
+
+template <typename KernelType>
+void launch_task(sycl::queue q, const KernelType& k);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.single_task(k)#.
+
+_Effects (2)_: Equivalent to [code]#h.submit([&](handler& h) { launch_task(h,
+k); })#.
+
+'''
+
+.[apititle]#launch_task# (kernel object)
+[source,role=synopsis,id=api:launch_task-kernel]
+----
+namespace sycl::khr {
+
+template <typename Args...>
+void launch_task(sycl::queue q, const sycl::kernel& k, Args&&... args);    (1)
+
+template <typename Args...>
+void launch_task(sycl::handler& h, const sycl::kernel& k, Args&&... args); (2)
+
+}
+----
+_Effects_: Enqueues a kernel object as a device task.
+The arguments in [code]#args# are passed to the kernel in the same order.
+
+'''
+
+=== Memory operations
+
+.[apititle]#memcpy#
+[source,role=synopsis,id=api:memcpy]
+----
+namespace sycl::khr {
+
+void memcpy(sycl::handler& h, void* dest, const void* src, size_t numBytes); (1)
+
+void memcpy(sycl::queue q, void* dest, const void* src, size_t numBytes);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.memcpy(dest, src, numBytes)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { memcpy(h, dest,
+src, numBytes); })#.
+
+'''
+
+.[apititle]#copy# (USM pointers)
+[source,role=synopsis,id=api:copy-pointer]
+----
+namespace sycl::khr {
+
+template <typename T>
+void copy(sycl::handler& h, const T* src, T* dest, size_t count); (1)
+
+template <typename T>
+void copy(sycl::queue q, const T* src, T* dest, size_t count);    (2)
+
+}
+----
+
+Copies between two USM pointers.
+
+_Constraints_: [code]#T# must be <<device-copyable>>.
```

Comment:
```suggestion
_Constraints_: [code]#T# is <<device-copyable>>.
```

I was looking recently at the C++ specification wording of _Constraints_, and they use the wording "is" instead of "must be".  For example, see the [containers library](https://eel.is/c++draft/containers).  I started adopting this wording style in #670 and the other "vec" PRs I drafted lately.

Reply:
Makes sense to me.  We should do a clean-up pass to get rid of all the "Available only if/when" constraints in SYCL 2020 and replace them with "is".